### ISSUE TITLE
Loss is NaN when using multiple GPUs

### ISSUE THREAD

**Issue by kwrobert** [minimaxir/gpt-2-simple#259]
Hello! 

First, great library. Made it super easy for me to dive in and start playing around with natural language processing models, so thank you! 

I'm trying to train the larger 774M model on a K80 by setting `multi_gpu=True` in the `finetune()` function. The output of `nvidia-smi` is clearly showing utilization on both GPUs in the card: 

```
Sun Feb 28 16:19:25 2021
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 450.102.04   Driver Version: 450.102.04   CUDA Version: 11.0     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  Tesla K80           Off  | 00000000:06:00.0 Off |                    0 |
| N/A   57C    P0    67W / 149W |   8360MiB / 11441MiB |     34%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   1  Tesla K80           Off  | 00000000:07:00.0 Off |                    0 |
| N/A   47C    P0    93W / 149W |   8368MiB / 11441MiB |    100%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   2  Quadro K2200        Off  | 00000000:82:00.0 Off |                  N/A |
| 42%   36C    P8     1W /  39W |    286MiB /  4040MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
```

and I no longer get the OOM errors I was seeing when running with `multi_gpu=False`. Great! 

However, the training loss is for some reason being returned as nan (this occurs for smaller model sizes as well). Output of a script running the `finetune()` function below:

```
Loading checkpoint models/774M/model.ckpt
Loading dataset...
dataset has 241968 tokens
Training...
2021-02-28 21:12:59.466015: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10.0
[1 | 92.45] loss=11.26 avg=11.26
[2 | 168.45] loss=nan avg=nan
[3 | 244.24] loss=nan avg=nan
[4 | 321.20] loss=nan avg=nan
[5 | 397.30] loss=nan avg=nan
[6 | 473.09] loss=nan avg=nan
```

This seems to be a bug of some sort, but I'm not sure what to make of it. Does anyone have any thoughts? Happy to share source code and training data if it would be helpful.

Thanks!
